File: 백준/Silver/9465. 스티커/스티커.py
from sys import *
from collections import * 
from heapq import * 
from functools import *
from math import *
from itertools import *

#setrecursionlimit(10**9)
input = stdin.readline


tc = int(input())

for _ in range(tc):
    n = int(input())
    score = []
    for _ in range(2):
        score.append([0] + list(map(int,input().split())))

    dp = [[0 for _ in range(n+1)] for _ in range(2)]
    
    dp[0][1] = score[0][1]
    dp[1][1] = score[1][1]
    
    for i in range(2,n+1):
        dp[0][i] = max(dp[1][i-2],dp[1][i-1]) + score[0][i]
        dp[1][i] = max(dp[0][i-1],dp[0][i-2]) + score[1][i]
    
    
    
    print(max(dp[0][n],dp[1][n]))

File: 백준/Silver/9465. 스티커/스티커.swift
import Foundation

var tc = Int(readLine()!)!

repeat {
    
    let n = Int(readLine()!)!
    var dp: [[Int]] = [[Int]](repeating: [Int](repeating: 0, count: n+1), count: 2)
    
    var arr :[[Int]] = []
    
    for _ in 0..<2 {
        arr.append([0] + readLine()!.split{$0 == " "}.map{Int($0)!})
    }
    
    for i in 1...n {
        
        dp[0][i] = max(dp[1][i-1] , ( i > 1 ? max(dp[0][i-2],dp[1][i-2]) : 0)) + arr[0][i]
        dp[1][i] = max(dp[0][i-1] , ( i > 1 ? max(dp[0][i-2],dp[1][i-2]) : 0)) + arr[1][i]
    }
    
    print(max(dp[0][n],dp[1][n]))
    
    
    tc -= 1
} while tc > 0
